unattended.sh

```diff
@@ -1,178 +1,615 @@
+	echo
+	echo "--"
+	echo " $script_short_name -h  == display this message"
+	echo
+	exit 1;
+}
+#======== end of function usage()
+
+
+#========= begin of function log()
+# print out the message according to the level
+# with timestamp
+# 
+# usage:
+# log "$LOG_LEVEL" "$message" ($LOG_LEVEL = DEBUG|INFO|WARNING|ERROR)
+#
+# example:
+# log "DEBUG" "This is a DEBUG_LOG_LEVEL message"
+# log "INFO" "This is a INFO_LOG_LEVEL message"
+#
+function log(){
+ TIMESTAMP=$(date --rfc-3339=seconds)
+ 
+ if [[ -z "${1}" || -z "${2}" ]]
+ then
+	echo "${TIMESTAMP} - ERROR : Missing argument"
+	echo "${TIMESTAMP} - ERROR : Usage log \"INFO\" \"Message log\" "
+	exit 1
+ fi
+ 
+ # get the message log level
+ log_message_level="${1}"
+ 
+ # get the log message
+ log_message="${2}"
+ 
+ # check if the log_message_level is greater than the runtime_log_level
+ [[ ${SUPPORTED_LOG_LEVEL[$log_message_level]} ]] || return 1
+ 
+ (( ${SUPPORTED_LOG_LEVEL[$log_message_level]} < ${SUPPORTED_LOG_LEVEL[$runtime_log_level]} )) && return 2
+ 
+ echo -e "${TIMESTAMP} - $log_message_level - $log_message" 
+
+}
+#======== end of function log()
+
+
+#========= begin of function parse_subcommand_options()
+# parse the provided arguments and check values
+# the script will display usage (and aborted) for any 
+# unsupported argument/option (which are defined in constants)
+# 
+function parse_subcommand_options(){
+	local requested_topology=""
+	local requested_version=""
+	local requested_repo=""
+	local OPTIND
+	OPTIND=2
+	while getopts $OPTIONS opt; do
+			case ${opt} in
+				t)
+					requested_topology=$OPTARG
+					log "INFO" "Requested topology   : '$requested_topology'"
+						
+					[[ ! ${SUPPORTED_TOPOLOGY[$requested_topology]} ]] && 
+						log "ERROR" "Unsupported topology : $requested_topology" &&
+						usage
+					;;
+					
+				v)
+					requested_version=$OPTARG
+					log "INFO" "Requested version    : '$requested_version'"
+					
+					[[ ! ${SUPPORTED_VERSION[$requested_version]} ]] && 
+						log "ERROR" "Unsupported version : $requested_version" &&
+						usage
+					;;
+					
+				r)
+					requested_repo=$OPTARG
+					log "INFO" "Requested repository : '$requested_repo'"
+					
+					[[ ! ${SUPPORTED_REPOSITORY[$requested_repo]} ]] && 
+						log "ERROR" "Unsupported repository : $requested_repo" &&
+						usage
+					;;
+					
+				l)
+					log_level=$OPTARG
+					if [ ! ${SUPPORTED_LOG_LEVEL[$log_level]} ];
+					then 
+						log "ERROR" "Unsupported and ignored log level : $log_level"
+					else
+						runtime_log_level=$log_level
+					fi
+					log "INFO" "Runtime log level set : $runtime_log_level"
+					;;
+					
+				\?) 
+					log "ERROR" "Invalid option: -"$OPTARG""
+				    usage
+					exit 1
+					;;
+					
+				: )
+					log "ERROR" "Option -"$OPTARG" requires an argument."
+					usage
+					exit 1
+					;;
+			esac
+	done
+    shift $((OPTIND-1))
```

Comment:
tabs everywhere instead of spaces
```suggestion
	shift $((OPTIND-1))
```